File: CppPractice/CppLeetcode/Leetcode141.cpp
class Solution {
public:
    bool hasCycle(ListNode *head) 
    {
        ListNode *fast=head;
        ListNode *slow=head;
        while(fast)
        {
            slow=slow->next;
            fast=fast->next;
            if(!fast){return false;}
            else{fast=fast->next;}
            if(slow==fast){return true;}
        }
        return false;
    }
};

File: CSharpPractice/LeetcodePractice/Leetcode141.cs
public class Solution141 {
    public bool HasCycle(ListNode head) 
    {
        HashSet<ListNode> visited = new HashSet<ListNode>();
        while(head!=null)
        {
            if(visited.Contains(head))
            {
                return true;
            }
            visited.Add(head);
            head =head.next;
        }
        return false;
    }
}